### PR TITLE
#161966915 User Profiles and Update

### PR DESCRIPTION
**Pull request by leonardnjura**
#### What does this PR do?
- Allows users to create their profile

#### How should this be manually tested?
`$ git clone -b ft-user-profiles-161966915 https://github.com/andela/ah-infinity-stones` then
`$ cd ah-infinity-stones`
> Create and activate virtual env
virtualenv -p python3 venv
`$ source venv/bin/activate`
`$ pip3 install -r requirements.txt to install the dependencies`
> Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice:
> Register user via `$ python3 manage.py createsuperuser --username john --email john.doe@example.com`

> `GET http://127.0.0.1:8000/api/profiles/`
> `GET http://127.0.0.1:8000/api/profiles/?page=2`
> `GET http://127.0.0.1:8000/api/profiles/u`
> `GET PUT http://127.0.0.1:8000/api/profiles/<username>`

```
export DATABASE_USER='user'
export DATABASE_PASSWORD='xxx'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='ah_test'

 export EMAIL_HOST_USER='sendgrid_username'
 export EMAIL_HOST_PASSWORD='sendgrid_pswd'
 
```
> Activate the env variables by `$source .env`
> Run migrations: $ python3 manage.py migrate
> Run generic tests - $ python3 manage.py test
> Run specific tests python3 manage.py test authors/apps/profiles

#### What are the relevant pivotal tracker stories?
[#161966915](https://www.pivotaltracker.com/story/show/161966915)

#### Screenshots
screen shot 2018-12-14 at 10 40 14
screen shot 2018-12-14 at 10 40 24
screen shot 2018-12-14 at 10 40 31
screen shot 2018-12-14 at 10 40 40

### CODE REVIEW

File: authors/apps/profiles/urls.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from django.urls import include, path
+from rest_framework import routers
+from rest_framework.urlpatterns import format_suffix_patterns
+
+# local imports
+from . import views
+
+router = routers.DefaultRouter()
+
+urlpatterns = [
```
Comment (codeplus254) [inline authors/apps/profiles/urls.py:11]: you ought to add the url path for confirmation email here as well :)

**Comment by rkemmy** [comment]
@leonardnjura the 'description of the task to be completed', 'what the PR does' and 'how to manually test this' are totally different and do not correlate to each other.

**Comment by rkemmy** [comment]
@leonardnjura please check email_confirm.html line 19 and make the necessary adjustments

**Comment by leonardnjura** [comment]
> @leonardnjura the 'description of the task to be completed', 'what the PR does' and 'how to manually test this' are totally different and do not correlate to each other.

**Comment by r-wambui** [comment]
@leonardnjura please work on the merge conflicts also rebase develop to pick the current changes.

**Comment by PromasterGuru** [comment]
Great work, work on the merge conflicts and rebase develope to ensure you have the latest changes.

File: authors/urls.py [hunk 2]
```diff
@@ -19,6 +19,9 @@
 urlpatterns = [
     path('', include('authors.apps.articles.url')),
     path('admin/', admin.site.urls),
-    path('api-auth/', include('rest_framework.urls', namespace='rest_framework')),
-    path('api/', include(('authors.apps.authentication.urls', 'authentication'), namespace='authentication')),
+    path('api-auth/', include(
+        'rest_framework.urls', namespace='rest_framework')),
+    path('', include('authors.apps.articles.url')),
+    path('api/profiles/', include('authors.apps.profiles.urls')),
+    path('api/', include('authors.apps.authentication.urls')),
```
Comment (r-wambui) [inline authors/urls.py]: @leonardnjura Ensure we have the same naming of the urls. Liaise with the team on this.

**Comment by r-wambui** [comment]
Resolve some of the tests, they are failing on travis but are passing develop travis build.

**Comment by leonardnjura** [comment]
@r-wambui note I fixed the code climate issues and cleaned up codebase as requested. Build passes locally. I also added a followers field as needed by @codeplus254. Feel free to go through once more before merge

File: authors/apps/authentication/templates/email_confirm.html [hunk 3]
```diff
@@ -1,6 +1,7 @@
 
     <!doctype html>
     <html>
+<<<<<<< HEAD
```
Comment (r-wambui) [inline authors/apps/authentication/templates/email_confirm.html]: @leonardnjura Why do we still have this?

Comment (leonardnjura) [reply authors/apps/authentication/templates/email_confirm.html]: Fixed it. Please check

**Comment by leonardnjura** [comment]
@r-wambui kindly check the changes requested on this PR

**Comment by ThaDeveloper** [comment]
To get or update a specific profile, is it `/profiles/u` or `profiles/<username>`? You could consider describing what the endpoints listed do.

**Comment by r-wambui** [comment]
@leonardnjura Please ensure that travis can complete the build to show the tests passing on this branch

File: authors/apps/profiles/urls.py (hunk 1 above)
Comment (leonardnjura) [reply authors/apps/profiles/urls.py:11]: Ok i will refactor

File: .travis.yml [hunk 4]
```diff
@@ -1,34 +1,47 @@
...
-  - coverage report
-
-after_success:
-  - coveralls
-
-notifications:
-  email: false
-
+# language: python
+# python:
```
Comment (r-wambui) [inline .travis.yml]: @leonardnjura Why did you comment this?

**Pull request merged**